libobs/obs.c

```diff
@@ -2375,6 +2375,9 @@ void obs_context_data_remove(struct obs_context_data *context)
 
 void obs_context_wait(struct obs_context_data *context)
 {
+	if (!context->mutex) {
```

Comment:
What's the context behind this change? Additionally, shouldn't we also make sure that the context is not null?

Reply:
_>> What's the context behind this change?_
`obs_source_destroy` is used to destroy a partially created source if a part of `obs_source_create_internal` fails. For example, when `obs_source_init` call fails, the source context already exits, so `obs_context_wait` will be called. It will crash because the `mutex `object does not exist yet.

_>> Additionally, shouldn't we also make sure that the context is not null?_
It is not a pointer.
```
struct obs_source {
	struct obs_context_data context;
```